lib/authorizeAccessToken.js

```diff
@@ -0,0 +1,187 @@
+/*!
+ * Copyright (c) 2021 Digital Bazaar, Inc. All rights reserved.
+ */
+import * as authorization from 'auth-header';
+import bedrock from 'bedrock';
+import * as JWT from '@digitalbazaar/minimal-jwt';
+import {tokenizers} from 'bedrock-tokenizer';
+import base64url from 'base64url-universal';
+
+const {util: {BedrockError}} = bedrock;
+
+/**
+ * Authorizes an incoming request for a route (parses Authorization: header,
+ * decodes access token, verifies it, checks the decoded token claims against
+ * the provided scope, sets the `req.clientMetadata` property).
+ *
+ * @param {object} options - Options hashmap.
+ * @param {string} options.scope - Scope required for the route guarded by this
+ *   `authorize()` usage.
+ * @param {Function} [options.customValidate] - Optional custom authorization
+ *   callback with the signature `async ({req, claims}) => {}`.
+ * @param {object} [options.cache] - Optional LRU cache to store the decoded
+ *   token claims). The cache API must be compatible with the lru-cache NPM
+ *   module.
+ * @param {string[]} options.validIssuers - List of allowed issuers of
+ *   access tokens (e.g. 'https://admin.sandbox.nacsav.com').
```

Comment:
sanitize this url, always use `example.com` in examples.

Reply:
Done 86edf10.